### PR TITLE
Alias for `i0` to `special` namespace

### PR DESCRIPTION
**Pull request by krshrimali**
See #50345 

cc: @mruberry @kshitij12345

### CODE REVIEW

**Comment by krshrimali** [comment]
Waiting for tests to pass, will request review once ready! (not marking as draft as I'm hoping it should be straight forward).

**Review by mruberry — APPROVED** [review]
Nice alias, @krshrimali!

**Comment by mruberry** [comment]
@krshrimali Unfortunately looks like this needs to be merged again

File: torch/csrc/api/include/torch/special.h [hunk 1]
```diff
@@ -182,6 +182,22 @@ inline Tensor& xlog1py_out(Tensor& result, const Tensor& self, const Scalar& oth
...
+/// Example:
+/// ```
+/// auto t = torch::randn(128, dtype=kDouble);
+/// torch::special::i0(t);
+/// ```
+inline Tensor i0(const Tensor& self) {
+  return torch::special_i0(self);
+}
+
+inline Tensor i0_out(const Tensor& self) {
```
Comment (kshitij12345) [inline torch/csrc/api/include/torch/special.h:197]: nit: `inline Tensor i0_out(const Tensor& self)` -> `inline Tensor& i0_out(Tensor& result, const Tensor& self)`

**Review by kshitij12345 — CHANGES_REQUESTED** [review]
Looks good except for one change.

Thanks!

File: torch/csrc/api/include/torch/special.h (hunk 1 above)
Comment (mruberry) [reply torch/csrc/api/include/torch/special.h:197]: Good catch, @kshitij12345; I always miss that ampersand

Comment (krshrimali) [reply torch/csrc/api/include/torch/special.h]: Thanks, @kshitij12345 - looks like ops like `ndtr, i0e, i1, i1e` are also missing the ampersand (`inline Tensor&`). Is it fine if I modify those in this PR? @mruberry @kshitij12345

Comment (mruberry) [reply torch/csrc/api/include/torch/special.h]: Fixing the out variants sounds good to me

Comment (kshitij12345) [reply torch/csrc/api/include/torch/special.h]: Ouch. Thanks! That will be great.

Comment (krshrimali) [reply torch/csrc/api/include/torch/special.h]: Thanks, should be resolved in the latest commit. :)

File: torch/csrc/api/include/torch/special.h [hunk 2]
```diff
@@ -182,6 +182,21 @@ inline Tensor& xlog1py_out(Tensor& result, const Tensor& self, const Scalar& oth
...
+/// ```
+/// auto t = torch::randn(128, dtype=kDouble);
+/// torch::special::i0(t);
+/// ```
+inline Tensor i0(const Tensor& self) {
+  return torch::special_i0(self);
+}
+
+inline Tensor& i0_out(Tensor& result, const Tensor& self) {
+  return torch::special_i0(result, self);
```
Comment (kshitij12345) [inline torch/csrc/api/include/torch/special.h:198]: nit: torch::special_i0_out

Comment (krshrimali) [reply torch/csrc/api/include/torch/special.h:198]: Apologies. Should be fixed now.

**Review by kshitij12345 — APPROVED** [review]
LGTM! Thanks for fixing the out function signature!

**Pull request closed without merging**